### PR TITLE
use envoy_google_grpc_external_deps for google_grpc_context_lib

### PR DESCRIPTION
**Pull request by qiwzhang**
Signed-off-by: Wayne Zhang <qiwzhang@google.com>

Directly use "external_deps" in google_grpc_context_lib causing 
source/exe:envoy_main_common_lib not able to support target:gce 

Description:
Risk Level: None
Testing: None
Docs Changes:
Release Notes:
[Optional Fixes #Issue]
[Optional Deprecated:]

### CODE REVIEW

**Comment by qiwzhang** [comment]
@jmarantz what do you think?

**Comment by mattklein123** [comment]
Please merge master to fix CI.

/wait

**Comment by qiwzhang** [comment]
It looks like the "bazel.release" test failed for this test:

//test/extensions/access_loggers/grpc:http_config_test 

In my local,. it passed "bazel test //test/extensions/access_loggers/grpc:http_config_test "

But I could not run "bazel.releae" in my local with docker.   I got this error:

./ci/run_envoy_docker.sh './ci/do_ci.sh bazel.release'
No remote cache bucket is set, skipping setup remote cache.
ENVOY_SRCDIR=/source
fatal: Not a git repository (or any parent up to mount point /build)
Stopping at filesystem boundary (GIT_DISCOVERY_ACROSS_FILESYSTEM not set).

**Comment by qiwzhang** [comment]
@jmarantz @htuch, @lizan  could you re-test it to see if this is a flaky test?

File: source/common/grpc/google_grpc_context.cc [hunk 1]
```diff
@@ -7,15 +7,19 @@
 #include "common/common/macros.h"
 #include "common/common/thread.h"
 
+#ifdef ENVOY_GOOGLE_GRPC
```
Comment (mattklein123) [inline source/common/grpc/google_grpc_context.cc]: Sorry why is this needed again? I thought we are not including this class unless Google gRPC is enabled? @jmarantz @lizan ?

Comment (jmarantz) [reply source/common/grpc/google_grpc_context.cc]: I did not understand why this was needed either. When I approved this PR this change had not been made yet.

**Comment by jmarantz** [comment]
Also @qiwzhang please don't force-push or squash commits. They drop history and comments made during review.

File: source/common/grpc/google_grpc_context.cc (hunk 1 above)
Comment (mattklein123) [reply source/common/grpc/google_grpc_context.cc]: @qiwzhang can you do a follow up to revert this part? Thank you.

Comment (qiwzhang) [reply source/common/grpc/google_grpc_context.cc]: Sorry,  #8407 will revert this change
